kms/keysource.go

```diff
@@ -278,7 +311,18 @@ func (key *MasterKey) NeedsRotation() bool {
 
 // ToString converts the key to a string representation.
 func (key *MasterKey) ToString() string {
-	return key.Arn
+	arnRole := key.Arn
+	if key.Role != "" {
+		arnRole = fmt.Sprintf("%s+%s", key.Arn, key.Role)
+	}
+	context := kmsContextToString(key.EncryptionContext)
+	if key.AwsProfile != "" {
+		return fmt.Sprintf("%s|%s|%s", arnRole, context, key.AwsProfile)
+	}
+	if len(key.EncryptionContext) > 0 {
```

Comment:
```suggestion
	if context != "" {
```
Just a thought, that the `kmsContextToString` could return something even if the length of the `EncryptionContext` is zero. It doesn’t right now, but maybe in the future?
Checking if the string is empty is also more consistent with the other if statements.

Reply:
I don't think `kmsContextToString()` should ever return anything than the empty string for an empty `EncryptionContext`.